src/test/smart-components/workflow/modals/add-workflow-modal-s.test.js

```diff
@@ -0,0 +1,150 @@
+    apiClientMock.get(`${APPROVAL_API_BASE}/workflows/?filter%5Bname%5D%5Bcontains_i%5D=&limit=50&offset=0`,
+      mockOnce({ body: { results: []}}));
+
+    jest.useFakeTimers();
+
+    const store = mockStore(initialState);
+    const wrapper = mount(
+      <ComponentWrapper store={ store }>
+        <Route path="/workflows/add-workflow/" render={ () => <AddWorkflow { ...initialProps } /> } />
+      </ComponentWrapper>
+    );
+
+    await act(async () => {
+      jest.advanceTimersByTime(550); // initial async validation
+    });
+    wrapper.update();
+
+    await act(async () => {
+      const name = wrapper.find('input').first();
+      name.instance().value = 'some-name';
+      name.simulate('change');
+    });
+    wrapper.update();
+
+    await act(async () => {
+      jest.advanceTimersByTime(550); // name async validaton
+    });
+    wrapper.update();
+
+    jest.useRealTimers();
+
+    await act(async () => {
+      const description = wrapper.find('textarea').first();
+      description.instance().value = 'some-description';
+      description.simulate('change');
+    });
+    wrapper.update();
+
+    wfHelper.addWorkflow = jest.fn().mockImplementation(() => Promise.resolve('ok'));
+
+    expect(wfHelper.addWorkflow).not.toHaveBeenCalled();
+
+    await act(async () => {
+      wrapper.find('form').simulate('submit');
+    });
+    wrapper.update();
+
+    expect(wfHelper.addWorkflow).toHaveBeenCalledWith({
+      group_refs: [],
+      name: 'some-name',
+      description: 'some-description'
+    });
+
+    expect(wrapper.find(MemoryRouter).instance().history.location.pathname).toEqual(routes.workflows.index);
+  });
+
+  it('should close the form', async () => {
+    localStorage.setItem('catalog_standalone', true);
+    localStorage.setItem('user', 'testUser');
+
+    apiClientMock.get(`${APPROVAL_API_BASE}/groups/?role=approval-approver&role=approval-admin`,
+      mockOnce({ body: { data: [{ uuid: 'id', name: 'name' }]}}));
+    apiClientMock.get(`${APPROVAL_API_BASE}/workflows/?filter%5Bname%5D%5Bcontains_i%5D=some-name&limit=50&offset=0`,
+      mockOnce({ body: { results: []}}));
+    apiClientMock.get(`${APPROVAL_API_BASE}/workflows/?filter%5Bname%5D%5Bcontains_i%5D=&limit=50&offset=0`,
+      mockOnce({ body: { results: []}}));
+
+    jest.useFakeTimers();
+
+    const store = mockStore(initialState);
+
+    const wrapper = mount(
+      <ComponentWrapper store={ store }>
+        <Route path="/workflows/add-workflow/" render={ () => <AddWorkflow { ...initialProps } /> } />
+      </ComponentWrapper>
+    );
+
+    await act(async () => {
+      jest.advanceTimersByTime(550); // initial async validation
+    });
+    wrapper.update();
+
+    jest.useRealTimers();
+
+    await act(async () => {
+      wrapper.find(Button).first().simulate('click');
+    });
+    wrapper.update();
+
+    expect(wrapper.find(MemoryRouter).instance().history.location.pathname).toEqual(routes.workflows.index);
```

Comment:
I'm not sure I understand the difference between this test and the one above. This same expectation is on line 112, and I don't see any other expectations in this test so it just seems like maybe a redundant test? Or was this meant to be a different expectation?

Reply:
One test is the path after a submit button click, the second is the return from a cancel button click.